packages/@aws-cdk/aws-codepipeline/lib/jenkins-provider.ts

```diff
@@ -0,0 +1,256 @@
+
+  _includeBuildVariant(): void;
+  _includeTestVariant(): void;
+
+  /**
+   * Convenience method for creating a new {@link JenkinsBuildAction},
+   * and adding it to the given Stage.
+   *
+   * @param stage the Pipeline Stage to add the new Action to
+   * @param name the name of the newly created Action
+   * @param props construction properties of the new Action
+   * @returns the newly created {@link JenkinsBuildAction}
+   */
+  addToPipeline(stage: cpapi.IStage, name: string, props: CommonJenkinsBuildActionProps):
+      JenkinsBuildAction;
+
+  /**
+   * Convenience method for creating a new {@link JenkinsTestAction},
+   * and adding it to the given Stage.
+   *
+   * @param stage the Pipeline Stage to add the new Action to
+   * @param name the name of the newly created Action
+   * @param props construction properties of the new Action
+   * @returns the newly created {@link JenkinsTestAction}
+   */
+  addToPipelineAsTest(stage: cpapi.IStage, name: string, props: CommonJenkinsTestActionProps):
+      JenkinsTestAction;
+}
+
+/**
+ * Properties for importing an existing Jenkins provider.
+ */
+export interface ImportJenkinsProviderProps {
+  /**
+   * The name of the Jenkins provider that you set in the AWS CodePipeline plugin configuration of your Jenkins project.
+   *
+   * @example 'MyJenkinsProvider'
+   */
+  providerName: string;
+
+  /**
+   * The base URL of your Jenkins server.
+   *
+   * @example 'http://myjenkins.com:8080'
+   */
+  serverUrl: string;
+
+  /**
+   * The version of your provider.
+   *
+   * @default '1'
+   */
+  version?: string;
+}
+
+export interface JenkinsProviderProps {
+  /**
+   * The name of the Jenkins provider that you set in the AWS CodePipeline plugin configuration of your Jenkins project.
+   *
+   * @example 'MyJenkinsProvider'
+   */
+  providerName: string;
+
+  /**
+   * The base URL of your Jenkins server.
+   *
+   * @example 'http://myjenkins.com:8080'
+   */
+  serverUrl: string;
+
+  /**
+   * The version of your provider.
+   *
+   * @default '1'
+   */
+  version?: string;
+
+  forBuild?: boolean;
+
+  forTest?: boolean;
+}
+
+export abstract class BaseJenkinsProvider extends cdk.Construct implements IJenkinsProvider {
+  public abstract readonly providerName: string;
+  public abstract readonly serverUrl: string;
+  public readonly version: string;
+
+  protected constructor(parent: cdk.Construct, id: string, version?: string) {
+    super(parent, id);
+
+    this.version = version || '1';
+  }
+
+  public export(): ImportJenkinsProviderProps {
+    return {
+      providerName: new cdk.Output(this, 'JenkinsProviderName', {
+        value: this.providerName,
+      }).makeImportValue().toString(),
+      serverUrl: new cdk.Output(this, 'JenkinsServerUrl', {
+        value: this.serverUrl,
+      }).makeImportValue().toString(),
+      version: new cdk.Output(this, 'JenkinsProviderVersion', {
+        value: this.version,
+      }).makeImportValue().toString(),
```

Comment:
You can export forBuild and forTest here as well.

Reply:
I could, but I'm not sure it makes much sense. For example, then `export` would be affected whether `addToPipeline` / `addToPipelineAsTest` was called before, which I'm not sure is the correct behavior.